src/hooks/useSafeInfo.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useMemo } from 'react'
+import { useQuery, type UseQueryResult } from '@tanstack/react-query'
+import type { Address } from 'viem'
+import type { ConfigParam, SafeConfig, SafeInfo } from '@/types/index.js'
+import { useConfig } from '@/hooks/useConfig.js'
+import { useSafeClient } from '@/hooks/useSafeClient.js'
+
+export type UseSafeInfoParams = ConfigParam<SafeConfig>
+export type UseSafeInfoReturnType = UseQueryResult<SafeInfo>
+
+export function useSafeInfo(params: UseSafeInfoParams = {}): UseSafeInfoReturnType {
+  const config = params.config ? useConfig({ config: params.config }) : useConfig()
+  const safeClient = useSafeClient(params.config ? { config: params.config } : undefined)
+
```

Comment:
The `config` constant already handles the config selection logic, so you don't need to repeat in `useSafeClient`

Reply:
Yes true! This looks better definitely.